libvuln/updates/manager.go

```diff
@@ -244,9 +248,43 @@ func (m *Manager) Run(ctx context.Context) error {
 	return nil
 }
 
+// updaterSetUpToDate checks for a stub updater set indicating that the whole updater set
+// is up to date and then be recorded the time in the db
+func (m *Manager) updaterSetUpToDate(ctx context.Context, set driver.UpdaterSet, updateTime time.Time) bool {
+	if len(set.Updaters()) == 1 {
+		if set.Updaters()[0].Name() == "rhel-all" {
+			updaterSet := "RHEL"
+			err := m.store.RecordUpdaterSetUpdateTime(ctx, updaterSet, updateTime)
+			if err != nil {
+				zlog.Error(ctx).
+					Err(err).
+					Str("updaterSet", updaterSet).
+					Str("updateTime", updateTime.String()).
+					Msg(fmt.Sprintf("error while recording update time for all updaters in updater set %s", updaterSet))
+			}
+			return true
+		}
+	}
+	return false
+}
+
 // DriveUpdater performs the business logic of fetching, parsing, and loading
 // vulnerabilities discovered by an updater into the database.
 func (m *Manager) driveUpdater(ctx context.Context, u driver.Updater) error {
+	var newFP driver.Fingerprint
```

Comment:
Oo I like, haven’t really used named returns so that was an interesting read. I may be misunderstanding your idea though as it seems in this case currently you don’t have to remember to assign to err for a return (you’re just have to explicitly decide what to return each time), whereas using a named variable would actually introduce making everyone to remember to assign to err (or at least remember to not use another name for a later error)

Reply:
Nice this is cool, I’ve changed it as you suggested which tidies the function up a little bit (possibly makes it a tad less readable though, I’m not sure).  
I had forgotten about the deferred function needing `err` though, so we still have to assign to it the whole way through unfortunately so not sure if it helps a great deal.